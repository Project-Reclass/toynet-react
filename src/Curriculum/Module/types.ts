export enum ModuleTypes {
  QUIZ = 'quiz',
  ARTICLE = 'article',
  PARENT = 'parent',
  EMULATOR = 'emulator',
  VALUE = 'value',
}

export interface ModuleInterface {
  id: number | string;
  moduleId: number | string;
  title: string;
  progress: number;
  type: ModuleTypes;
}