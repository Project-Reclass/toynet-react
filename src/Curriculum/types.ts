export interface Module {
  id: number;
  name: string;
  introduction: string;
  submodules: string;
}

export interface SubModule {
  id: number;
  name: string;
  introduction: string;
  type: 'SURVEY' | 'VALUE' | 'LESSON' | 'ARTICLE' | 'LAB';
}