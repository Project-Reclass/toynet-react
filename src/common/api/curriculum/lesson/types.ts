export interface LessonMetaRequest {
  id: number;
}