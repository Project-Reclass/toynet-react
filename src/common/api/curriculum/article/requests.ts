/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import axios from 'axios';

export interface ArticleImage {
  file: string;
  credit: string;

  caption?: string;
}

export interface ArticleMeta {
  source: string;
  title: string;
  author: string;
  images: ArticleImage[];
}

export const getArticleText = async (articleId: number): Promise<string> => {
  const { data } = await axios.get(`/data/article/${articleId}/article.md`);
  return data;
};

export const getArticleMeta = async (articleId: number): Promise<ArticleMeta> => {
  const { data } = await axios.get(`/data/article/${articleId}/meta.json`);
  return data;
};