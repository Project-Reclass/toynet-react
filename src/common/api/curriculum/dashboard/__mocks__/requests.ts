/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/

import { DashboardIntf } from 'src/common/types/curriculum';

const defaultMockData = {
  id: 1,
  name: 'test',
  introduction: 'intro',
  modules: [],
};

export let mockData: DashboardIntf = {...defaultMockData};

export function __resetMock() {
  mockData = {...defaultMockData};
}

export function __setMockData(data: DashboardIntf) {
  mockData.id = data.id;
  mockData.name = data.name;
  mockData.introduction = data.introduction;
  mockData.modules = data.modules;
}

export const getCurriculum = async (): Promise<DashboardIntf | undefined> => {
  return Promise.resolve(mockData);
};