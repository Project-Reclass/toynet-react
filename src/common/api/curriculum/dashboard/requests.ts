import axios from 'axios';
import { DashboardIntf } from 'src/common/types/curriculum';

export const getCurriculum = async (curriculumId: number): Promise<DashboardIntf | undefined> => {
  const { data } = await axios.get(`/data/curriculum/${curriculumId}.json`);
  return data;
};