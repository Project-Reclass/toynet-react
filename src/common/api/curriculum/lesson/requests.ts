import axios from 'axios';

type LessonSlideURI = string;
export interface Meta {
  numSlides: number;
}

export const getLessonSlide = async (submoduleId: number, slideId: number): Promise<LessonSlideURI> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/${slideId}.png`);
  return data;
};

export const getLessonMeta = async (submoduleId: number): Promise<Meta | undefined> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/meta.json`);
  return data;
};
