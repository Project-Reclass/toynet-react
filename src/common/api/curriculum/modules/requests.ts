import { simulateNetwork } from 'src/common/utils';
import modules from 'src/__data__/curriculum/modules.json';

export const getAllModules = () => {
  return simulateNetwork(() => modules);
};

export const getModuleById = (id: number) => {
  return simulateNetwork(() => modules.find(module => module.id === id));
};

export const getModuleByName = (name: string) => {
  return simulateNetwork(() => modules.find(module => module.name === name));
};