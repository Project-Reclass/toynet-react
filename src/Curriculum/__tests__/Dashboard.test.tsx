/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/


import React from 'react';

import ModuleList from 'src/Curriculum';
import { renderWithTheme } from 'src/common/test-utils/renderWithTheme';
import RenderWithRouter from 'src/common/test-utils/renderWithRouter';
import { waitFor } from '@testing-library/react';
import * as requests from 'src/common/api/curriculum/dashboard/requests';
import { DashboardIntf } from 'src/common/types/curriculum';
import { asMockedModule } from 'src/common/test-utils/types';

jest.mock('src/common/api/curriculum/dashboard/requests');

const path = '/dashboard/:curriculumId';
const initialEntries = ['/dashboard/1'];

const mockRequest = asMockedModule<DashboardIntf, typeof requests>(requests);


describe('The Module List', () => {
  afterEach(() => {
    mockRequest.__resetMock();
  });
  it('should render and match snapshot', async () => {
    const { container, getByText } = renderWithTheme(
      <RenderWithRouter
        path='/dashboard/:curriculumId'
        initialEntries={[`/dashboard/${1}`]}
      >
        <ModuleList />
      </RenderWithRouter>,
    );

    await waitFor(() => getByText(/intro/i));

    expect(container).toMatchSnapshot();
  });

  it('should render the introduction of the curriculum', async () => {
    const mockedName = 'new cool introduction';
    mockRequest.__setMockData({
      ...mockRequest.mockData,
      introduction: mockedName,
    });
    const { getByText } = renderWithTheme(
      <RenderWithRouter
        path={path}
        initialEntries={initialEntries}
      >
        <ModuleList />
      </RenderWithRouter>,
    );

    await waitFor(() => getByText(new RegExp(mockedName, 'i')));

    expect(getByText(new RegExp(mockedName, 'i'))).toBeInTheDocument();
  });

  it('should render the modules for the curriculum', async () => {
    mockRequest.__setMockData({
      ...mockRequest.mockData,
      modules: [
        {
          id: 1,
          name: 'First module name',
          introduction: 'first module intro',
          submodules: [],
        },
      ],
    });
    const { getByText } = renderWithTheme(
      <RenderWithRouter
        path={path}
        initialEntries={initialEntries}
      >
        <ModuleList />
      </RenderWithRouter>,
    );

    await waitFor(() => getByText(new RegExp('first module intro', 'i')));

    expect(getByText(/first module intro/i)).toBeInTheDocument();
    expect(getByText(/first module name/i)).toBeInTheDocument();
  });

});
