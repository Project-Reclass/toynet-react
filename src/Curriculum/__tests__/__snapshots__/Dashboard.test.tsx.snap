// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Module List should render and match snapshot 1`] = `
<div>
  <div
    class="css-1oqmqkw"
  >
    <div
      class="css-1djaw1e"
    >
      <div
        class="css-mo2z4l"
      >
        <div
          class="chakra-stack css-zihvxp"
        >
          <h2
            class="chakra-heading css-r0uwm9"
          >
            Hi there!
          </h2>
          <p
            class="chakra-text css-1c395im"
          >
            intro You can click on each moduleâ€™s chevron to see its submodules. You must go through
                the modules and their submodules in order. To start or revisit a submodule, click 
            <span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="display: inline;"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
            </span>
             button.
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-t5uauj"
    />
  </div>
</div>
`;
